features/aggregating_failures.feature

```diff
@@ -0,0 +1,97 @@
+@wip @announce
+Feature: Aggregating Failures
+
+  Normally, an expectation failure causes the example to immediately bail.  Sometimes, however, you have multiple, independent expectations, and you'd like to be able to see all of the failures rather than just the first.  One solution is to split off a separate example for each expectation, but if the setup for the examples is slow, that's going to take extra time and slow things down. `aggregate_failures` provides an alternate solution. Within the block, expectation failures will not abort the example like normal; instead, the failures will be aggregated into a single exception that is raised at the end of the block, allowing you to see all expectations that failed.
+
+  `aggregate_failures` takes an optional string argument that will be used in the aggregated failure message as a label.
+
+  Scenario: Multiple Expectation Failures in the same example are all reported
```

Comment:
I think we need both.  In most tests that do one action and than may want to assert multiple things, wrapping the entire test works well and is really convenient.  However, in an end-to-end acceptance test, you can have multiple actions (e.g. simulating a user clicking through your app and doing stuff) and you won't want later steps to run if something failed at an earlier step...but _within_ a step it's nice to be able to assert multiple things.

You're right that this isn't explained well; I'll improve it...although, I'm still not sure where the cukes belong given that it sits at the intersection of core and expectations.

On a side note, I'm thinking about recommending to the cucumber team that in their rspec-expectations integration they may want to wrap every step in an `aggregation_failures` block.